asdf/tests/helpers.py

```diff
@@ -184,7 +184,7 @@ def _assert_roundtrip_tree(
     tree_match_func="assert_equal",
 ):
 
-    fname = str(tmpdir.join("test.asdf"))
+    fname = str(tmp_path / "test.asdf")
```

Comment:
Since this will be used for testing other packages (e.g. `astropy`), we need to confirm this does not change the interface. Does this work still if one passes a `tmpdir` fixture?

Reply:
Thanks for pointing that out. I made a small change in [b7edb3a](https://github.com/asdf-format/asdf/pull/1217/commits/b7edb3a71a965ed738ddfb356331853db19a9826) (convert tmp_path to string and us os.path.join) to support both.